### ISSUE TITLE
SPIDEV0 not working on Beaglebone

### ISSUE THREAD

**Issue by FRVisser** [resin-os/resinos#197]
Here there, 

We need the spidev0 port on the Beaglebone. Normally you could `echo BB-SPIDEV0 > /sys/devices/platform/bone_capemgr/slots` and the cape gets loaded. However, on Resin this doesn't seem to work. I tried on both Resin OS 1.19 and Resin OS 2.0.0-beta-7. 

We use the same command for UART on other projects, and this works fine for a long time already. Our problem seems to be specific on the spidev0 port. 

I have already seen https://forums.resin.io/t/how-do-you-enable-the-spi-pins/242. I tried this but unfortunately doesn't work.

Andrei also pointed me at:  

```
RobertCNelson commented on May 5, 2016
For spi, you need to have them loaded via the kernel parameter in /boot/uEnv.txt

echoing it to the slots while in userspace is too late..
```
Issue on [adafruit-beaglebone-io-python](https://github.com/adafruit/adafruit-beaglebone-io-python/issues/94#issuecomment-217075941) library ( tried this one too)

I have tried this on both `BeagleBoard.org Debian Image 2016-11-06` (kernel `4.4.30-ti-r64`) and `BeagleBoard.org Debian Image 2015-11-12` (kernel `3.8.13-bone79`). And echoing SPIDEV works in userlevel. I see an immediate pin state change, and if I send something trough `/dev/spidev1.0`, my logic analyser receives it. 

Now the resin part. 
If I  echo spidev there are no errors. Cape manager shows the cape is loaded and the port shows up as `/dev/spidev1.0` or `spidev1.1`. When I point my program to this port, no errors there either. Only giving zero's as received values. 

As it is now, there is no way to load a cape from the uEnv.txt.

**Comment by FRVisser** [comment]
Any updates? Could you guys reproduce it?

**Comment by agherzan** [comment]
Hello @FRVisser . We didn't managed to take a look on this. If you by any chance figure it out before we pick it up, do share. Thanks for understanding.

**Comment by FRVisser** [comment]
Getting back to this issue since I both have trouble with using UART5 and also using SPI1.
How do I start tackling these problems? I have been searching and testing a couple of things. 

In the latest official Beaglebone Debian image, the capemgr is ditched completely and all setting moved to the `uEnv.txt` file. 

I got some feedback this time. If I now echo :

`echo BB-SPIDEV1 > /sys/devices/platform/bone_capemgr/slots`

I get error messages:

```   
[ 1007.020008] bone_capemgr bone_capemgr: part_number 'BB-SPIDEV1', version 'N/A'
[ 1007.028167] bone_capemgr bone_capemgr: slot #7: override
[ 1007.043140] bone_capemgr bone_capemgr: Using override eeprom data at slot 7
[ 1007.059134] bone_capemgr bone_capemgr: slot #7: 'Override Board Name,00A0,Override Manuf,BB-SPIDEV1'
[ 1007.096160] pinctrl-single 44e10800.pinmux: pin PIN100 already requested by 48038000.mcasp; cannot claim for 481a0000.spi
[ 1007.151282] pinctrl-single 44e10800.pinmux: pin-100 (481a0000.spi) status -22
[ 1007.194603] pinctrl-single 44e10800.pinmux: could not request pin 100 (PIN100) from group pinmux_bb_spi1_pins  on device pinctrl-single
[ 1007.235124] omap2_mcspi 481a0000.spi: Error applying setting, reverse things back
[ 1007.252613] omap2_mcspi: probe of 481a0000.spi failed with error -22
[ 1007.264142] bone_capemgr bone_capemgr: slot #7: dtbo 'BB-SPIDEV1-00A0.dtbo' loaded; overlay id #1
```

 `cat /sys/devices/platform/bone_capemgr/slots ` shows:

```
 0: PF----  -1 
 1: PF----  -1 
 2: PF----  -1 
 3: PF----  -1 
 4: P-O-L-   0 Override Board Name,00A0,Override Manuf,BB-SPIDEV1
```

`mcasp@48038000` is  `compatible = "ti,omap2-mcasp-audio";` some sort of audio device, that clearly needs to be disabled in uEnv.txt

This brings up the first question: How do I set the `uEnv.txt` configuration in Resin? There is now a `resinOS_uEnv.txt` but I can't seem to find it, nor do I know what the base of it is.

**Comment by FRVisser** [comment]
In contact with the Resin support we have found a solution:

The HDMI cape needs to be disabled by in the `uEnv.txt` file. After this, I was able to use and test `BB-SPIDEV0`, `BB-SPIDEV1`, `BB-UART1`, `BB-UART2`, `BB-UART4` and `BB-SPIDEV5`.

By appending `fdtfile=am335x-boneblack-emmc-overlay.dtb` to the `uEnv.txt_internal` file on `flash-boot` before flashing with SD-card or `uEnv.txt` file on `resin-boot` (`/dev/mmcblk1p1`) after flashing. 

Combined with echo-ing the slots to the capemgr. Like:
```echo BB-SPIDEV1 > /sys/devices/platform/bone_capemgr/slots```
